Dropdown styling

RGRMdesign opened the pull request:
Hi @YOUR1 

In deze PR:

- Styling voor dropdowns 
- Hiermee kan straks de extra dropdown voor Tools werken.
- Bug met niet zichtbaar zijn van de dropdown bij het gebruikersmenu
- Styling update voor tooltips. Let op, ik heb hierbij wat aanpassingen gedaan in de icon-only.js en in de less bestanden

RGRMdesign committed:
add icons

RGRMdesign committed:
add styling for dropdowns

RGRMdesign committed:
update dist files

RGRMdesign committed:
update dist files

RGRMdesign committed:
move and update tooltip

RGRMdesign committed:
remove extra z-index declaration

RGRMdesign committed:
update dist files

YOUR1 commented:
Welk icoon kan ik gebruiken voor de tool menu? Ik zie die hier niet zo snel in staan.

RGRMdesign commented:
Hi @YOUR1,

Die zit er nog niet in. Ik zal 'm toevoegen in deze PR, ik noem 'm `ti-tools`.

Kun je mij het overzicht van de id's geven die in de dropdown komen te staan? Dan voeg ik in deze PR ook daarvoor de juiste iconen toe.

Groet,
Rogier

YOUR1 commented:
Ik zal mijn wijzigingen ook in deze codebase zetten zodat we 1 alles omvattende repo hebben.

HTML van tools:
```HTML
<li class="tools-item navigation-item with-background with-dropdown navigation-item-icon-only expanded">
   <a href="#" class="navigation-link" aria-haspopup="true" aria-expanded="true" aria-controls="toolsMenu" title="Tools menu">
   <span class="visually-hidden">Tools menu</span>
   <i class="ti ti-tools"></i>
   </a>
   <ul class="navigation-item-dropdown-menu align-left" role="menu" aria-hidden="false" id="toolsMenu">
      <li id="t-whatlinkshere" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:VerwijzingenNaarHier/NORA2:Help" title="Lijst met alle pagina's die naar deze pagina verwijzen [ctrl-option-j]" accesskey="j" class="navigation-link">Verwijzingen naar deze pagina</a></li>
      <li id="t-upload" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:Uploaden" title="Bestanden uploaden [ctrl-option-u]" accesskey="u" class="navigation-link">Bestand uploaden</a></li>
      <li id="t-specialpages" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:SpecialePaginas" title="Lijst met alle speciale pagina's [ctrl-option-q]" accesskey="q" class="navigation-link">Speciale pagina's</a></li>
      <li id="t-permalink" class="mw-list-item navigation-item"><a href="/index.php?title=NORA2:Help&amp;oldid=75732" title="Permanente koppeling naar deze versie van deze pagina" class="navigation-link">Permanente koppeling</a></li>
      <li id="t-smwbrowselink" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:Bekijken/:NORA2:Help" rel="search" class="navigation-link">Eigenschappen bekijken</a></li>
   </ul>
</li>
```

Let wel op: gaat nog niet helemaal goed met styling.
<img width="118" alt="Screenshot 2025-03-12 at 09 38 10" src="https://github.com/user-attachments/assets/1cddfd3a-7d2a-4fd8-a8e1-9809d325bab8" />

Verder heb ik de HTML ook klaa
<img width="297" alt="Screenshot 2025-03-12 at 09 38 46" src="https://github.com/user-attachments/assets/616efca1-c884-4fa7-898a-7efe6916037a" />
r staan voor de naamruimte switcher


HTML
```HTML
<div class="namespace-choice-container">
    <input type="radio" id="radio-ca-nstab-project" name="namespace-choice" class="namespace-choice-radio" value="ca-nstab-project" data-href="/index.php/NORA2:Help" checked="" />
    <label for="radio-ca-nstab-project" class="namespace-choice-label" title="Projectpagina bekijken [a]">Projectpagina</label>
    <input type="radio" id="radio-ca-talk" name="namespace-choice" class="namespace-choice-radio" value="ca-talk" data-href="/index.php?title=Overleg_NORA2:Help&amp;action=edit&amp;redlink=1" />
    <label for="radio-ca-talk" class="namespace-choice-label" title="Overleg over deze pagina (de pagina bestaat niet) [t]">Overleg</label>
</div>
```

YOUR1 committed:
Adding Tool actions

YOUR1 committed:
Refactor namespace choice and icon portlet JavaScript

RGRMdesign commented:
Nice die namespace switcher! Ik had dit in deze PR ook al meegenomen, maar dan nog zonder dat het radio-buttons waren. Dat zal ik dus nog even ombouwen naar jouw implementatie. Kun je me dan alleen nog laten zien hoe die `namespace-choice-container` in de html van de dropdown komt te staan?

YOUR1 commented:
> Nice die namespace switcher! Ik had dit in deze PR ook al meegenomen, maar dan nog zonder dat het radio-buttons waren. Dat zal ik dus nog even ombouwen naar jouw implementatie. Kun je me dan alleen nog laten zien hoe die `namespace-choice-container` in de html van de dropdown komt te staan?

De gehele HTML van de tools ziet er op dit moment zo uit:
```HTML
<ul class="tools">
    <li id="ca-edit" class="tools-item navigation-item navigation-item-icon-only mw-list-item"><a href="/index.php?title=Informatie&amp;action=edit" class="tool-edit navigation-link" title="Bron bewerken"><span class="visually-hidden">Bron bewerken</span><i class="ti ti-edit"></i></a></li>
    <li id="ca-formedit" class="tools-item navigation-item navigation-item-icon-only mw-list-item"><a href="/index.php?title=Informatie&amp;action=formedit" class="tool-formedit navigation-link" title="Bewerken"><span class="visually-hidden">Bewerken</span><i class="ti ti-formedit"></i></a></li>
    <li id="ca-watch" class="tools-item navigation-item navigation-item-icon-only mw-list-item"><a href="/index.php?title=Informatie&amp;action=watch" class="tool-watch navigation-link" data-mw="interface" title="Volgen"><span class="visually-hidden">Volgen</span><i class="ti ti-watch"></i></a></li>
    <li id="ca-history" class="tools-item navigation-item navigation-item-icon-only mw-list-item"><a href="/index.php?title=Informatie&amp;action=history" class="tool-history navigation-link" title="Geschiedenis weergeven"><span class="visually-hidden">Geschiedenis weergeven</span><i class="ti ti-history"></i></a></li>

    <li class="tools-item navigation-item smartcomments-toggler with-background with-dropdown">
        <a href="#" class="navigation-link" aria-haspopup="true" aria-expanded="false" aria-controls="smartCommentsMenu">
                        SmartComments <span class="status">Uit</span>
                    </a>
        <ul class="navigation-item-dropdown-menu align-left" role="menu" id="smartCommentsMenu" aria-hidden="true">
            <li class="navigation-item sic-enable-commenting" id="ca-comments" role="none">
                <a class="navigation-link" href="#">Aan</a>
            </li>
        </ul>
    </li>
    <li class="tools-item navigation-item with-background with-dropdown navigation-item-icon-only expanded">
        <a href="#" class="navigation-link" aria-haspopup="true" aria-expanded="true" aria-controls="toolsMenu" title="Tools menu">
            <span class="visually-hidden">Tools menu</span>
            <i class="ti ti-tools"></i>
        </a>
        <ul class="navigation-item-dropdown-menu align-left" role="menu" aria-hidden="false" id="toolsMenu">
            <li id="t-whatlinkshere" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:VerwijzingenNaarHier/Informatie" title="Lijst met alle pagina's die naar deze pagina verwijzen [ctrl-option-j]" accesskey="j" class="navigation-link">Verwijzingen naar deze pagina</a></li>
            <li id="t-upload" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:Uploaden" title="Bestanden uploaden [ctrl-option-u]" accesskey="u" class="navigation-link">Bestand uploaden</a></li>
            <li id="t-specialpages" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:SpecialePaginas" title="Lijst met alle speciale pagina's [ctrl-option-q]" accesskey="q" class="navigation-link">Speciale pagina's</a></li>
            <li id="t-permalink" class="mw-list-item navigation-item"><a href="/index.php?title=Informatie&amp;oldid=70655" title="Permanente koppeling naar deze versie van deze pagina" class="navigation-link">Permanente koppeling</a></li>
            <li id="t-smwbrowselink" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:Bekijken/:Informatie" rel="search" class="navigation-link">Eigenschappen bekijken</a></li>

        </ul>
    </li>
    <li class="tools-item navigation-item with-background with-dropdown navigation-item-icon-only">
        <a href="#" class="navigation-link" aria-haspopup="true" aria-expanded="false" aria-controls="contentMenu" title="Overige pagina handelingen bekijken">
            <span class="visually-hidden">Overige pagina handelingen bekijken</span>
            <i class="ti ti-dots"></i>
        </a>
        <ul class="navigation-item-dropdown-menu align-left" role="menu" aria-hidden="true" id="contentMenu">
            <li class="navigation-item info">Naamruimten</li>
            <div class="namespace-choice-container">
                <input type="radio" id="radio-ca-nstab-main" name="namespace-choice" class="namespace-choice-radio" value="ca-nstab-main" data-href="/index.php/Informatie" checked="">
                <label for="radio-ca-nstab-main" class="namespace-choice-label" title="Inhoudspagina bekijken [c]">Pagina</label>
                <input type="radio" id="radio-ca-talk" name="namespace-choice" class="namespace-choice-radio" value="ca-talk" data-href="/index.php?title=Overleg:Informatie&amp;action=edit&amp;redlink=1">
                <label for="radio-ca-talk" class="namespace-choice-label" title="Overleg over deze pagina (de pagina bestaat niet) [t]">Overleg</label>
            </div>
            <li class="navigation-item info">Meer</li>
            <li id="ca-delete" class="mw-list-item navigation-item"><a href="/index.php?title=Informatie&amp;action=delete" title="Deze pagina verwijderen [ctrl-option-d]" accesskey="d" class="navigation-link">Verwijderen</a></li>
            <li id="ca-move" class="mw-list-item navigation-item"><a href="/index.php/Speciaal:PaginaHernoemen/Informatie" title="Deze pagina hernoemen [ctrl-option-m]" accesskey="m" class="navigation-link">Hernoemen</a></li>
            <li id="ca-protect" class="mw-list-item navigation-item"><a href="/index.php?title=Informatie&amp;action=protect" title="Deze pagina beveiligen [ctrl-option-=]" accesskey="=" class="navigation-link">Beveiligen</a></li>
            <li id="ca-smartbooks" class="mw-list-item navigation-item"><a href="https://nora2.dev139.wikixl.nl/index.php?title=Informatie&amp;action=smartbook" class="navigation-link">Pagina exporteren</a></li>
            <li id="ca-purge" class="mw-list-item navigation-item"><a href="/index.php?title=Informatie&amp;action=purge" class="navigation-link">Vernieuwen</a></li>

        </ul>
    </li>
</ul>
```

RGRMdesign commented:
Thanks! Zou je om de `<div class="namespace-choice-container">` nog een `<li>` kunnen zetten ofzo? Anders is de html niet valide en kunnen er dingen gebeuren die we niet willen...

YOUR1 commented:
> Thanks! Zou je om de `<div class="namespace-choice-container">` nog een `<li>` kunnen zetten ofzo? Anders is de html niet valide en kunnen er dingen gebeuren die we niet willen...

Ah nice catch, had ik niet gezien. Ga ik oppakken 👍

YOUR1 committed:
Improve namespace choice styling and HTML generation

- Wrap portlet choice container in a list item
- Modify namespace choice container border and width
- Update radio button checked state color
- Center and adjust label styling
- Remove XML encoding declaration from HTML generation

RGRMdesign committed:
add tools icon

RGRMdesign committed:
add icon classes

RGRMdesign committed:
update html and js

RGRMdesign committed:
style tools button and dropdown

RGRMdesign committed:
update namespace switch styling

RGRMdesign committed:
show tools button on small vw

RGRMdesign committed:
Merge branch 'dropdown-styling' of https://github.com/NORArchitectuur/mw-skin into dropdown-styling

RGRMdesign commented:
Hi @YOUR1 

Ik heb e.e.a. aangepast en gefikst. Zie commits hierboven. 2 aandachtspunten:

- Ik heb de styling die ik al had voor de Pagina/Overleg links aangepast naar de nieuwe html met de radio buttons. Ik heb `namespace-choice.less` verwijderd en de verwijzing in `skin.json` verwijderd.
- Hoe willen we omgaan met Edit, Volgen, etc. op mobiel? Afgesproken is dat deze op mobiel te vinden zijn onder de 3 dots dropdown. Zet je deze erbij met JS of zullen ze er altijd staan en verbergen we ze met css bij een grotere viewport width?

RGRMdesign committed:
update dist files

YOUR1 commented:
> Hi @YOUR1
> 
> Ik heb e.e.a. aangepast en gefikst. Zie commits hierboven. 2 aandachtspunten:

Top, ik zal de branch mergen en zodadelijk uitrollen op de staging omgeving

> * Ik heb de styling die ik al had voor de Pagina/Overleg links aangepast naar de nieuwe html met de radio buttons. Ik heb `namespace-choice.less` verwijderd en de verwijzing in `skin.json` verwijderd.
> * Hoe willen we omgaan met Edit, Volgen, etc. op mobiel? Afgesproken is dat deze op mobiel te vinden zijn onder de 3 dots dropdown. Zet je deze erbij met JS of zullen ze er altijd staan en verbergen we ze met css bij een grotere viewport width?

Ik moet hier nog over nadenken. Het helemaal optimaal krijgen voor mobiel zal in een aparte branch gebeuren, ik verwacht dat ik dat deels ga doen met CSS/JS.
